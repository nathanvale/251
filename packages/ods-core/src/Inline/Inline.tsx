import React, { Children } from "react";
import {
  ResponsiveSpace,
  CollapsibleAlignmentProps,
  SpaceVariants,
  ComponentBaseProps,
} from "@origin-digital/ods-types";
import flattenChildren from "react-keyed-flatten-children";
import {
  cssLengthToString,
  mapToStyledSystem,
} from "@origin-digital/ods-helpers";
import styled, { css } from "styled-components";
import { style, get, ResponsiveValue } from "styled-system";
import { resolveCollapsibleAlignmentProps } from "../_private/resolveCollapsibleAlignmentProps";
import { Box } from "../Box";

export interface InlineProps
  extends CollapsibleAlignmentProps,
    ComponentBaseProps {
  space?: ResponsiveSpace;
  children: React.ReactNode;
}

interface OuterContainerProps {
  space: ResponsiveValue<SpaceVariants>;
}

interface InnerContainerProps {
  space: ResponsiveValue<SpaceVariants>;
}

const preventCollapse = 1;

const base = css`
  padding-top: ${preventCollapse}px;
  &:before {
    content: "";
    display: block;
  }
`;

const marginTop = style({
  prop: "space",
  cssProperty: "margin-top",
  key: "space",
  transformValue: (n, scale) =>
    cssLengthToString(0 - parseInt(get(scale, n), 10) - preventCollapse),
});

const marginLeft = style({
  prop: "space",
  cssProperty: "margin-left",
  key: "space",
  transformValue: (n, scale) =>
    cssLengthToString(0 - parseInt(get(scale, n), 10)),
});

const negativeMarginTop = css`
  &:before {
    ${marginTop}
  }
`;

const negativeMarginLeft = css`
  ${marginLeft}
`;

const OuterContainer = styled(Box)<OuterContainerProps>`
  ${base}
  ${negativeMarginTop}
`;
export const InnerContainer = styled(Box)<InnerContainerProps>`
  min-width: 100%;
  ${negativeMarginLeft}
`;
const ChildContainer = styled(Box)`
  min-width: 0;
`;

const defaultSpace = "none";

export const Inline = ({
  "data-id": dataId,
  children,
  space = defaultSpace,
  alignX,
  alignY,
  collapseBelow,
  ...rest
}: InlineProps) => {
  const {
    collapsibleAlignmentProps,
    collapsibleAlignmentChildProps,
  } = resolveCollapsibleAlignmentProps({
    alignX,
    alignY,
    collapseBelow,
  });

  return (
    <OuterContainer
      space={mapToStyledSystem(space) || "none"}
      data-id={dataId}
      {...rest}
    >
      <InnerContainer
        space={mapToStyledSystem(space) || "none"}
        flexWrap="wrap"
        {...collapsibleAlignmentProps}
      >
        {Children.map(flattenChildren(children), (child) =>
          child !== null && child !== undefined ? (
            <ChildContainer
              paddingLeft={space}
              paddingTop={space}
              {...collapsibleAlignmentChildProps}
            >
              {child}
            </ChildContainer>
          ) : null
        )}
      </InnerContainer>
    </OuterContainer>
  );
};

Inline.displayName = "Inline";
Inline.defaultProps = {
  "data-id": "inline",
  space: "none",
  alignX: "left",
  alignY: "top",
};
