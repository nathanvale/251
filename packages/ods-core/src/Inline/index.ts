export { Inline, InnerContainer } from "./Inline";
export type InlineProps = import("./Inline").InlineProps;
