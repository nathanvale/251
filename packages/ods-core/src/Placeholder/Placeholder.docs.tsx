import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Placeholder, PlaceholderProps } from "./Placeholder";

export const docs: ComponentDocs<PlaceholderProps> = {
  category: "Layout",
  componentName: "Placeholder",
  description:
    "This component is helpful for laying out placeholders for future content.",
  propDescriptions: {},
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <Placeholder />,
    },
    additional: [
      {
        label: "Setting the Label",
        Code: () => <Placeholder label="Placeholder for content" />,
      },
      {
        label: "Setting the Shape",
        Code: () => <Placeholder shape="round" width={120} height={120} />,
      },
      {
        label: "Setting the Width",
        Code: () => <Placeholder width={120} />,
      },
      {
        label: "Setting the Height",
        Code: () => <Placeholder height={400} />,
      },
      {
        label: "Responsive Height Across Mobile and Desktop",
        description:
          "Width can be controlled responsively by passing in an array of mobile and desktop pixel sizes as values. Open in Playroom to see the responsive behaviour of this example.",
        Code: () => <Placeholder height={[100, 500]} />,
      },
      {
        label: "Responsive Height Across All Five Breakpoints",
        description:
          "For more granular control across all 5 breakpoints, ‘xs’, ‘sm’,  ‘md’, ‘lg’ and ‘xl’, width can be controlled responsively by passing in an object with breakpoints as keys and pixel sizes as values. Open in Playroom to see the responsive behaviour of this example.",
        Code: () => (
          <Placeholder
            height={{
              xs: 100,
              sm: 200,
              md: 300,
              lg: 400,
              xl: 500,
            }}
          />
        ),
      },
    ],
  },
  snippets: [],
};
