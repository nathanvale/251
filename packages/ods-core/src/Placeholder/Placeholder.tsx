import React from "react";
import styled, { css } from "styled-components";
import { width, height, WidthProps, HeightProps } from "styled-system";
import { ResponsiveProp } from "@origin-digital/ods-types";
import { normaliseResponsiveProp } from "@origin-digital/ods-helpers";
import { Box, BoxProps } from "../Box/Box";

type PlaceholderShapeVariant = "rectangle" | "round";
export interface PlaceholderProps {
  "data-id"?: string;
  height?: ResponsiveProp<string | number>;
  label?: string;
  shape?: PlaceholderShapeVariant;
  width?: ResponsiveProp<string | number>;
}

interface ContainerProps extends WidthProps, HeightProps {
  shape: PlaceholderProps["shape"];
}

const Container = styled<Omit<BoxProps, "width" | "height">>(Box)<
  ContainerProps
>`
  ${width}
  ${height}
  border: 2px solid rgba(51, 51, 51, 0.3);
  ${p =>
    p.shape === "round"
      ? css`
          border-radius: 50%;
        `
      : undefined}}
`;

const defaultHeight = 120;
const defaultWidth = "auto";
const defaultShape = "rectangle";

export const Placeholder = ({
  "data-id": dataId,
  label,
  width = defaultWidth,
  height = defaultHeight,
  shape = defaultShape,
}: PlaceholderProps) => {
  return (
    <Container
      data-id={dataId}
      position="relative"
      overflow="hidden"
      display="flex"
      alignItems="center"
      justifyContent="center"
      width={normaliseResponsiveProp<string | number>(width)}
      height={normaliseResponsiveProp<string | number>(height)}
      shape={shape}
      backgroundColor="grey16"
    >
      {label ? (
        <Box
          paddingX="xsmall"
          paddingY="xxsmall"
          textAlign="center"
          style={{ marginTop: "2px" }}
        >
          {label}
        </Box>
      ) : (
        <svg
          xmlns="http://www.w3.org/2000/svg"
          style={{ width: "100%", height: "100%", position: "absolute" }}
        >
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1={0}
            y1={0}
            x2="100%"
            y2="100%"
          />
          <line
            style={{ strokeWidth: "2px", stroke: "rgba(51,51,51,.1)" }}
            x1="100%"
            y1={0}
            x2={0}
            y2="100%"
          />
        </svg>
      )}
    </Container>
  );
};

Placeholder.defaultProps = {
  "data-id": "placeholder",
  height: defaultHeight,
  inline: false,
  shape: defaultShape,
  width: defaultWidth,
};

Placeholder.displayName = "Placeholder";
