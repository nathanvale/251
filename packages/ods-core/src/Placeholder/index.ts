export { Placeholder } from "./Placeholder";
export type PlaceholderProps = import("./Placeholder").PlaceholderProps;
