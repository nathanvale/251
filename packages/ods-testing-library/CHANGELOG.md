# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
