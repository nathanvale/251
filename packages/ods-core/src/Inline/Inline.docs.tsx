import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Placeholder } from "../Placeholder";
import { Inline, InlineProps } from "./Inline";

export const docs: ComponentDocs<InlineProps> = {
  category: "Layout",
  componentName: "Inline",
  description:
    "Renders content horizontally with consistent spacing between all items, spanning multiple lines if needed.",
  propDescriptions: {
    alignX: "This is a responsive prop which aligns the children horizontally.",
    alignY: "This is a responsive prop which aligns the children vertically.",
    collapseBelow:
      "Switch from horizontal layout to vertical below the selected breakpoint.",
    space:
      "The responsive prop T-shirt sized space between all items, spanning multiple lines if needed.",
  },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <Inline space="small">
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
          <Placeholder height={48} width={48} />
        </Inline>
      ),
    },
    additional: [
      {
        label: "Responsive space at mobile and desktop, ['xxsmall', 'medium']",
        Code: () => (
          <Inline space={["xxsmall", "medium"]}>
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
          </Inline>
        ),
      },
      {
        label: "Align horizontally to center",
        Code: () => (
          <Inline space="small" alignX="center">
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
          </Inline>
        ),
      },
      {
        label: "Align horizontally to right",
        Code: () => (
          <Inline space="small" alignX="right">
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
          </Inline>
        ),
      },
      {
        label:
          "Responsive alignment (center on mobile, left from desktop upwards)",
        Code: () => (
          <Inline space="small" alignX={["center", "left"]}>
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
          </Inline>
        ),
      },
      {
        label: "Align vertically",
        Code: () => (
          <Inline space="small" alignY="center">
            <Placeholder width={48} height={40} />
            <Placeholder width={48} height={100} />
            <Placeholder width={48} height={60} />
          </Inline>
        ),
      },
      {
        label: "Collapse below desktop",
        Code: () => (
          <Inline space="small" collapseBelow="lg">
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
          </Inline>
        ),
      },
      {
        label:
          'Collapse below desktop with responsive space ( "xxsmall" on mobile, "large" on desktop)',
        Code: () => (
          <Inline space={["xxsmall", "large"]} collapseBelow="lg">
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
          </Inline>
        ),
      },
      {
        label: "Collapse below desktop with center alignment",
        Code: () => (
          <Inline space="small" collapseBelow="lg" alignX="center">
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
          </Inline>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Small space",
      Code: () => (
        <Inline space="small">
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
        </Inline>
      ),
    },
    {
      label: "Medium space",
      Code: () => (
        <Inline space="medium">
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
        </Inline>
      ),
    },
    {
      label: "Responsive space",
      Code: () => (
        <Inline space={["small", "large"]}>
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
        </Inline>
      ),
    },
    {
      label: "Responsive horizontal alignment",
      Code: () => (
        <Inline space="small" alignX={["center", "left"]}>
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
          <Placeholder width={48} height={48} />
        </Inline>
      ),
    },
    {
      label: "Vertically centered",
      Code: () => (
        <Inline space="small" alignY="center">
          <Placeholder width={48} height={40} />
          <Placeholder width={48} height={100} />
          <Placeholder width={48} height={60} />
        </Inline>
      ),
    },
  ],
};
