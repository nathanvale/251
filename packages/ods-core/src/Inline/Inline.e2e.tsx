/* eslint-disable react/display-name */
import React, { Fragment } from "react";
import { E2ETests } from "@origin-digital/ods-types";
import { Placeholder } from "../Placeholder/Placeholder";
import { Inline } from "./Inline";

export const tests: E2ETests = [
  {
    label:
      "It should CollapseBelow + align: On mobile should be vertical and left aligned,  on desktop should be horizontal and right aligned",
    Code: () => (
      <Inline space="small" collapseBelow="lg" alignX={["left", "right"]}>
        <Placeholder width={48} height={48} />
        <Placeholder width={48} height={48} />
        <Placeholder width={48} height={48} />
        <Placeholder width={48} height={48} />
        <Placeholder width={48} height={48} />
        <Placeholder width={48} height={48} />
      </Inline>
    ),
    responsive: true,
  },
  {
    label:
      "It should flatten fragments (6 placeholders should be evenly spaced)",
    Code: () => (
      <Inline space="small">
        <Fragment>
          <Fragment>
            <Placeholder width={48} height={48} />
          </Fragment>
          <Fragment>
            <Placeholder width={48} height={48} />
          </Fragment>
        </Fragment>
        <Fragment>
          <Fragment>
            <Placeholder width={48} height={48} />
            <Placeholder width={48} height={48} />
            <Fragment>
              <Placeholder width={48} height={48} />
            </Fragment>
          </Fragment>
        </Fragment>
        <Placeholder width={48} height={48} />
      </Inline>
    ),
    responsive: true,
  },
];
