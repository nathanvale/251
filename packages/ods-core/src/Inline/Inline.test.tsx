import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Placeholder } from "../Placeholder";
import { Inline } from "./Inline";

test("it renders", () => {
  const { container } = render(
    <Inline space="small">
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
    </Inline>
  );
  expect(container.firstChild).toMatchSnapshot();
});
