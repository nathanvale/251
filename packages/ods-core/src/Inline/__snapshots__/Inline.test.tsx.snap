// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.c3 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c5 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  display: block;
  padding-top: 12px;
  padding-left: 12px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c7 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  background-color: #e3e3e3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  overflow: hidden;
}

.c0 {
  padding-top: 1px;
}

.c0:before {
  content: "";
  display: block;
}

.c0:before {
  margin-top: -13px;
}

.c2 {
  min-width: 100%;
  margin-left: -12px;
}

.c4 {
  min-width: 0;
}

.c6 {
  width: 48px;
  min-width: 10px;
  height: 48px;
  border: 2px solid rgba(51,51,51,0.3);
}

@media screen and (min-width:576px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:992px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:1200px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:576px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:992px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:1200px) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:576px) {
  .c3 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:992px) {
  .c3 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:1200px) {
  .c3 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:576px) {
  .c5 {
    display: block;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    display: block;
  }
}

@media screen and (min-width:992px) {
  .c5 {
    display: block;
  }
}

@media screen and (min-width:1200px) {
  .c5 {
    display: block;
  }
}

@media screen and (min-width:576px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:992px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media screen and (min-width:1200px) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<div
  class="c0 c1"
  data-id="inline"
>
  <div
    class="c2 c3"
    display="flex,flex,flex,flex,flex"
  >
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4 c5"
      display="block,block,block,block,block"
    >
      <div
        class="c6 c7"
        data-id="placeholder"
        display="flex"
        height="48"
        overflow="hidden"
        shape="rectangle"
        width="48"
      >
        <svg
          style="width: 100%; height: 100%; position: absolute;"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="0"
            x2="100%"
            y1="0"
            y2="100%"
          />
          <line
            style="stroke-width: 2px; stroke: rgba(51,51,51,.1);"
            x1="100%"
            x2="0"
            y1="0"
            y2="100%"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
